### ISSUE TITLE
Period Predicate "Error Case" Spec compliance.

### ISSUE THREAD

**Issue by mbutlerw** [xtdb/crux#2299]
> If one or more <period predicand>s specify PERIOD, then let m be the number of <period predicand>s specified. For i, 1 (one) ≤ i ≤ m, let PSVi be the value of <period start value> in the i-th <period predicand> and let PEVi be the value of <period end value> in the i-th <period predicand>.
> a) If at least one of PSVi and PEVi, 1 (one) ≤ i ≤ m, is the null value, then the result of the specified <period overlaps predicate>, <period equals predicate>, <period contains predicate>, <period precedes predicate>, <period succeeds predicate>, <period immediately precedes predicate>, or <period immediately succeeds predicate> is Unknown and no further General Rules of this Subclause are applied.
> b) If for any i, 1 (one) ≤ i ≤ m, PSVi ≥ PEVi, then an exception condition is raised: data exception invalid period value.

> TLDR:
> if any start or end period value is NULL, return unknown
> If any start is greater than or equal to end raise runtime data exception.

This is problematic for us as we split the individual clauses of the predicate up to push them around, meaning there's no natural point to compare the start and end. Possibly we need to introduce a guard clause which throws the aforementioned exception.
